### PR TITLE
[4.0] - RavenDB-11267 & RavenDB-11304

### PR DESCRIPTION
**Pull request by grisha-kotler**
http://issues.hibernatingrhinos.com/issue/RavenDB-11267
http://issues.hibernatingrhinos.com/issue/RavenDB-11304

### CODE REVIEW

File: src/Raven.Studio/wwwroot/App/views/shell/registration.html [hunk 1]
```diff
@@ -46,15 +46,25 @@ <h3 class="modal-title" id="myModalLabel">
...
-                <button type="submit" data-bind="css: { 'btn-spinner': isBusy }, enable: !isBusy()" class="btn btn-primary">Submit</button>
+                <!-- ko if: canForceUpdate -->
+                <button class="btn btn-info" data-bind="click: forceLicenseUpdate, disable: spinners.forceLicenseUpdate,
+                                    css: { 'btn-spinner': spinners.forceLicenseUpdate }">
+                    <i class="icon-force"></i>
+                    <span>Force Update</span>
+                </button>
+                <!-- /ko -->
+
+                <button type="submit" data-bind="css: { 'btn-spinner': isBusy }, enable: !isBusy(), disable: spinners.forceLicenseUpdate"
```
Comment (ml054) [inline src/Raven.Studio/wwwroot/App/views/shell/registration.html]: you can't use both enable and disable bindings together.

File: src/Raven.Studio/typescript/viewmodels/shell/registration.ts [hunk 2]
```diff
@@ -92,6 +98,10 @@ class registration extends dialogViewModelBase {
             return firstStart.diff(now, "days");
         });
 
+        this.canForceUpdate = ko.pureComputed(() => {
+            return licenseStatus.Expired || !!this.error();
```
Comment (ml054) [inline src/Raven.Studio/typescript/viewmodels/shell/registration.ts:102]: can license status be null ?

File: src/Raven.Studio/typescript/viewmodels/shell/registration.ts [hunk 3]
```diff
@@ -147,6 +157,23 @@ class registration extends dialogViewModelBase {
...
+                    .done(() => {
+                        const licenseStatus = license.licenseStatus();
+                        if (!licenseStatus.Expired &&
+                            !licenseStatus.ErrorMessage) {
+                            app.closeDialog(this);
+                        }
+                        license.fetchSupportCoverage();
+                    });
+            })
+            .always(() => this.spinners.forceLicenseUpdate(false));
```
Comment (ml054) [inline src/Raven.Studio/typescript/viewmodels/shell/registration.ts:175]: any failure/ success message here?

**Review by ml054 — CHANGES_REQUESTED** [review]
Please have a look at my notes.

**Comment by ayende** [comment]
Looks fine for me in terms of `.cs` changes

**Comment by ppekrol** [comment]
@grisha-kotler can `ReloadLicense` method execute `LicenseChanged?.Invoke();` in `tryfinally` ? I think we need to inform certain compontents of the system that license changed even when we fallback to AGPL

**Comment by grisha-kotler** [comment]
@ppekrol we already invoke LicenseChanged when there is a license.
It is impossible to get back to AGPL (no license) once you installed a license.
You can only update/change your license (in that case LicenseChanged will be invoked)

**Comment by ayende** [comment]
@grisha-kotler This conflicts

**Comment by grisha-kotler** [comment]
fixing it

File: src/Raven.Studio/wwwroot/App/views/shell/registration.html (hunk 1 above)
Comment (grisha-kotler) [reply src/Raven.Studio/wwwroot/App/views/shell/registration.html]: done

File: src/Raven.Studio/typescript/viewmodels/shell/registration.ts (hunk 2 above)
Comment (grisha-kotler) [reply src/Raven.Studio/typescript/viewmodels/shell/registration.ts:102]: No (even if there is no license).

File: src/Raven.Studio/typescript/viewmodels/shell/registration.ts (hunk 3 above)
Comment (grisha-kotler) [reply src/Raven.Studio/typescript/viewmodels/shell/registration.ts:175]: will be shown by the forceLicenseUpdateCommand

**Comment by grisha-kotler** [comment]
Can be merged

**Pull request merged**